mitmproxy/optmanager.py

```diff
@@ -1,21 +1,49 @@
 import contextlib
 import blinker
 import pprint
+import inspect
+import copy
+import functools
+import weakref
+import os
+
+import ruamel.yaml
 
 from mitmproxy import exceptions
 from mitmproxy.utils import typecheck
 
+
 """
     The base implementation for Options.
 """
 
 
-class OptManager:
+class _DefaultsMeta(type):
+    def __new__(cls, name, bases, namespace, **kwds):
+        ret = type.__new__(cls, name, bases, dict(namespace))
+        defaults = {}
+        for klass in reversed(inspect.getmro(ret)):
+            for p in inspect.signature(klass.__init__).parameters.values():
+                if p.kind in (p.KEYWORD_ONLY, p.POSITIONAL_OR_KEYWORD):
```

Comment:
The POSITIONAL_OR_KEYWORD type is caught here so we don't fail in mysterious ways when keyword-only forcing was forgotten somewhere, possibly in a base class.